src/test/java/com/github/dockerjava/core/command/CreateContainerCmdImplTest.java

```diff
@@ -561,4 +563,59 @@ public void createContainerWithLogConfig() throws DockerException {
         // null becomes empty string
         assertEquals(inspectContainerResponse.getHostConfig().getLogConfig().type, logConfig.type);
     }
+
+    /**
+     * https://github.com/calavera/docker/blob/3781cde61ff10b1d9114ae5b4c5c1d1b2c20a1ee/integration-cli/docker_cli_run_unix_test.go#L319-L333
+     */
+    @Test
+    public void testWithStopSignal() throws Exception {
+        Integer signal = 10; // SIGUSR1 in busybox
+
+        CreateContainerResponse resp = dockerClient.createContainerCmd(BUSYBOX_IMAGE)
+                .withCmd("/bin/sh", "-c", "trap 'echo \"exit trapped 10\"; exit 10' USR1; while true; do sleep 1; done")
+                .withAttachStdin(true)
+                .withTty(true)
+                .withStopSignal(signal.toString())
+                .exec();
+        final String containerId = resp.getId();
+        assertThat(containerId, not(isEmptyString()));
+        dockerClient.startContainerCmd(containerId).exec();
+
+        InspectContainerResponse inspect = dockerClient.inspectContainerCmd(containerId).exec();
+        assertThat(inspect.getState().getRunning(), is(true));
+
+        dockerClient.stopContainerCmd(containerId).exec();
+        Thread.sleep(TimeUnit.SECONDS.toMillis(3));
+
+        inspect = dockerClient.inspectContainerCmd(containerId).exec();
+        assertThat(inspect.getState().getRunning(), is(false));
+        assertThat(inspect.getState().getExitCode(), is(signal));
+
+        StringBuilder stringBuilder = new StringBuilder();
+        final StringBuilderLogReader callback = new StringBuilderLogReader(stringBuilder);
+        dockerClient.logContainerCmd(containerId)
+                .withStdErr(true)
+                .withStdOut(true)
+                .withTailAll()
+                .exec(callback);
+//                .awaitCompletion(); FIXME https://github.com/docker-java/docker-java/issues/476
```

Comment:
throws NegativeArraySizeException as described in #476. The fix is to check if `read` returns > -1